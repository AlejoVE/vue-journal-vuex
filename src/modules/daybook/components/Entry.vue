<template>
  <div class="entry-container mb-3 pointer p-2" @click="$router.push({name: 'entry', params: { id: 10 }} )">
      <div class="entry-title d-flex">
          <span class="text-success fs-5 fw-bold">1</span>
          <span class="mx-1 fs-5">DEC</span>
          <span class="mx-2 fw-light">2021</span>
      </div>
      <div class="entry-description">
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor vitae sunt accusantium illo voluptatum itaque eligendi necessitatibus maiores provident asperiores velit quis quasi reiciendis qui, obcaecati ipsam quisquam consequuntur exercitationem?
      </div>
  </div>
</template>

<script>
export default {

}
</script>

<style lang="scss" scoped>
.entry-container {
    border-bottom: 1px solid #2c3e50;
    transition: 0.2s all ease-in;

    &:hover {
        background-color: lighten($color: grey, $amount: 45);
        transition: 0.2s all ease-in;
    }
    .entry-description {
        font-size: 12px;
    }
}

</style>