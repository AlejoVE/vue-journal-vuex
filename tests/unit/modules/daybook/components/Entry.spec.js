import Entry from '@/modules/daybook/components/Entry'
import {shallowMount} from '@vue/test-utils'
import journalState from '../../../mocks-data/test-journal-state'

describe('Tests in Entry Component', () => {
    const mockRouter = {
        push: jest.fn()
    }

    const wrapper = shallowMount(Entry, {
        global: {
            mocks: {
                $router: mockRouter
            }
        },
        props: {
            entry: journalState.entries[0]
        }
    })

    test('should match snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    test('should redirect when click in entry-container', () => {
        const entryContainer =  wrapper.find('.entry-container')

        entryContainer.trigger('click')

        expect(mockRouter.push).toHaveBeenCalled()
        expect(mockRouter.push).toHaveBeenCalledWith({ name: 'entry', params: {id: journalState.entries[0].id} })
    })

    test('Tests in computed properties', () => {
        const {day, month, yearDay} = wrapper.vm

        expect(day).toBe(13)
        expect(month).toBe('Dec')
        expect(yearDay).toBe('2021, Mon')
    })

})
