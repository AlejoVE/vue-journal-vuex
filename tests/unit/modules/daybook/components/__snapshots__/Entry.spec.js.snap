// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in Entry Component should match snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">13</span><span class="mx-1 fs-5">Dec</span><span class="mx-2 fw-light">2021, Mon</span></div>
  <div class="entry-description">this is my first entry in vuex test</div>
</div>
`;
